### PR TITLE
feat(ts-sdk): convert .js extension file to .ts

### PR DESCRIPTION
**Pull request by TomChv**
This is an attempt to move forward #4368, to do so, we need code to use `.ts` import extension so it can be handled by the Deno Runtime.

This PR fixes that issue so we'll only have to create a `deno.json` file, some runtime adjustment & we should be good to go.

### CODE REVIEW

**Review by jedevc — COMMENTED** [review]
Just to confirm, existing modules that use `.js` or `.ts` source files will keep working right?

File: cmd/codegen/generator/typescript/templates/src/header.ts.gtpl [hunk 1]
```diff
@@ -7,8 +7,8 @@ inherited by futures objects and common types.
  * This file was auto-generated by `client-gen`.
  * Do not make direct changes to the file.
  */
-import { Context, defaultContext } from "../context/context.js"
-import { computeQuery } from "./utils.js"
+import { Context, defaultContext } from "../context/context.ts"
+import { computeQuery } from "./utils.ts"
```
[the remark this answers is not in the document]
Comment (TomChv) [reply cmd/codegen/generator/typescript/templates/src/header.ts.gtpl:11]: Linter

File: sdk/typescript/runtime/bin/__tsconfig.updator.ts [hunk 2]
```diff
@@ -65,5 +67,14 @@ if (
...
   ]
 
+  // Add noEmit & allowImportingTsExtensions if they are not part of the tsconfig
+  if (!tsconfig.compilerOptions.noEmit) {
+    tsconfig.compilerOptions.noEmit = true
+  }
+
+  if (!tsconfig.compilerOptions.allowImportingTsExtensions) {
+    tsconfig.compilerOptions.allowImportingTsExtensions = true
+  }
```
[the remark this answers is not in the document]
Comment (TomChv) [reply sdk/typescript/runtime/bin/__tsconfig.updator.ts:77]: They are to not emit .js file and allowing import with .ts extension notation.

**Comment by TomChv** [comment]
> Just to confirm, existing modules that use .js or .ts source files will keep working right?

They should, the option simply allow it, it doesn't mean it will error on .js extensio, specially since tsx (our ts code runner) supports both

File: sdk/typescript/runtime/bin/__tsconfig.updator.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jedevc) [reply sdk/typescript/runtime/bin/__tsconfig.updator.ts:77]: Right, I understand what the options do. Specifically, `noEmit`, why does this one need to be set?

[the remark this answers is not in the document]
Comment (TomChv) [reply sdk/typescript/runtime/bin/__tsconfig.updator.ts:77]: Because `tsconfig.compilerOptions.allowImportingTsExtensions` cannot be set otherwise

**Comment by TomChv** [comment]
No release note since this is a code refactor.

**Pull request merged**